### PR TITLE
Update discovery client JSON service path

### PR DESCRIPTION
**Pull request by lschroed**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?
Updates the discovery client to look for the discovery service JSON at `NISHAREDDIR64/Services/Services.json`.

### Why should this Pull Request be merged?
The path for the discovery service JSON is being changed, the client needs to look for it in the new location.

### What testing has been done?
Ran `pytest` locally.

### CODE REVIEW

**Comment by lschroed** [comment]
@jonlugoatx, could you confirm the updated path is correct?

**Comment by jonlugoatx** [comment]
> @jonlugoatx, could you confirm the updated path is correct?

Yes, that's correct.

**Review by jonlugoatx — APPROVED** [review]

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (dixonjoel) [inline ni_measurement_plugin_sdk/discovery/_support.py:61]: I don't see why we need this fallback. We plan to update the minimum required version of InstrumentStudio to 2024 Q3 in the docs, so I don't think we expect anyone to try our next release with older InstrumentStudio / MeasurementLink. @rfriedma ?

**Review by dixonjoel — APPROVED** [review]
Approved with suggestion to remove the fallback.

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (rfriedma) [reply ni_measurement_plugin_sdk/discovery/_support.py:61]: presuming the developer has 2024 Q3+ installed then it would seem a fallback isn't required

Comment (bkeryan) [reply ni_measurement_plugin_sdk/discovery/_support.py:61]: You are also renaming the package, so it's not like existing applications will automatically upgrade.

**Review by bkeryan — APPROVED** [review]

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (lschroed) [reply ni_measurement_plugin_sdk/discovery/_support.py]: I've removed the fallback with the latest update.

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (jasonmreding) [inline ni_measurement_plugin_sdk/discovery/_support.py:1]: > Updates the discovery client to look for the discovery service JSON at NISHAREDDIR64/Services/Services.json first. If that does not exist, then try to use the legacy MeasurementLink JSON path.

I don't see any fallback to the old path here. I'm not sure this is entirely needed unless we want to support the measurement running against older versions of InstrumentStudio. If the test suite is using older installations of our services, then this will definitely be needed.

Comment (lschroed) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: I removed the fallback based on @dixonjoel's suggestion. Is that not correct to remove? I've updated the PR description to reflect that change.

Comment (jasonmreding) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: I don't know that it is strictly needed. I just know that it's different than the [C# client](https://dev.azure.com/ni/DevCentral/_git/ASW?path=/Source/MeasurementServices/DiscoveryClient/ServicesConfigurationFileLocator.cs&version=GBmain&line=74&lineEnd=84&lineStartColumn=1&lineEndColumn=18&lineStyle=plain&_a=contents). @jonlugoatx is there anything I'm missing where you know the fallback is required?

For Python, I guess if you weren't using a virtual environment and somebody updated to the latest release of this package for the system, then it might break existing python based measurements if an older version of InstrumentStudio was installed. I know that's not a deployment strategy we would recommend though.

Comment (bkeryan) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: @jasonmreding 

> [bkeryan](https://github.com/bkeryan) [2 hours ago](https://github.com/ni/measurement-plugin-python/pull/752#discussion_r1640237247)
> You are also renaming the package, so it's not like existing applications will automatically upgrade.

Comment (jasonmreding) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: That's true. I forgot about those pending PRs in the context of this change. So I guess it is really a question of whether or not we care about supporting older versions of IS with the latest release of this repo or not. And more specifically, whether we care about supporting dev work flows for measurements where the measurement is started manually from the command line rather than from IS or TS.

Comment (rfriedma) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: we're moving the minimum requirements for this repo to IS 2024 Q3. I'm more concerned with existing measurements built on measurementlink continuing to be runnable without error where possible. If needed, I'd accept the scenario that old measurements can no longer automatically start the discovery service when running from the dev workflow. Hopefully the upgrade path is easy enough that developers can easily fix the dev workflow by simply upgrading to latest.

**Comment by bkeryan** [comment]
BTW @lschroed please update the PR title so it isn't just your dev branch name.

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (jonlugoatx) [inline ni_measurement_plugin_sdk/discovery/_support.py:60]: After discussing with Ryan Eckenrode, we decided to change this to `NIServices.json` since searching the ancestor directory hierarchy for Services.json from the C# client is too generic and might pick up some JSON file we don't own.

**Review by jonlugoatx — CHANGES_REQUESTED** [review]

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (lschroed) [reply ni_measurement_plugin_sdk/discovery/_support.py]: Updated to `NIServices.json`.

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (jasonmreding) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: @rfriedma If that's the case, then why bother to still support V1 measurements. It seems like we could slowly clean some things up if we're not going to support older clients any longer. I guess there is still the question about what to do with older TestStand clients.

Comment (rfriedma) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: I'm not sure I described that well then. Old measurements should keep working from TS and IS. My understanding is that Measurement Steps and IS both launch the Discovery Service. I was reading this as "there's potential that in dev workflows, the discovery service won't automatically get launched with the latest version of this SDK" which seemed like a fairly contained slice that doesn't impact older measurements running from TS and IS.

**Review by jonlugoatx — APPROVED** [review]

File: ni_measurement_plugin_sdk/discovery/_support.py
Comment (lschroed) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: Should we add the fallback path back in then? That doesn't seem to add risk and allows for later dropping support of old clients.

Comment (jasonmreding) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: I'll defer to code owners, but if it is easy and it is consistent with our clients in other languages, then I would say yes.

Comment (bkeryan) [reply ni_measurement_plugin_sdk/discovery/_support.py:1]: Restoring the fallback path is fine with me.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**
